### ISSUE TITLE
Adding in_place kwarg, to transition to returning the modified DataStore object

### ISSUE THREAD

**Issue by BSchilperoort** [dtscalibration/python-dts-calibration#126]
Like pandas used to do, most methods of `DataStore` change the object in place, and do not return a new object. In xarray the object methods do not modify the original object, but return a modified version.

Pros of inplace=False:
- Easier to try different calibration kwarg inputs without changing the original `DataStore` object. This avoids the need for explicitly making copies of the `DataStore`, or reloading the data from disk.
- The way pandas and xarray does things; could be more convenient for new users
- Consistent between methods (eg. `mean` returns a new object, while `calibration_single_ended` modifies in place)
- Allows method chaining

Pros of inplace=True
- The way we currently do it
- Method chaining isn't useful for calibration

What do you think? Should we add an `in_place` kwarg, and eventually deprecate that in favor of `in_place=False` being the default? Or just leave things like they are now.

**Comment by bdestombe** [comment]
I would vote for inplace=False, and try to follow the guidelines of xarray as much as possible. Note that Dask requires inplace=False. Thus on Dataset level, they work inplace=True, but on DataArray level, they are work with inplace=False, right? 

But for this to work, you'd have to split the input from the output. I don't think anyone want the Stokes measurements twice in their memory. What do you mean with "Method chaining isn't useful for calibration"?

If I compare two calibration strategies, I construct the two DataStores from scratch, from netCDF files. To me, that is more explicit and little extra work. But I am curious, do you have like an example of an use case?

**Comment by BSchilperoort** [comment]
You can find an example of method chaining here: https://www.dataschool.io/future-of-pandas/

E.g.:
```python
ds = (dtscalibration.read_silixa_files(files)
                    .sel(x=slice(0, 100)
                    .calibration_single_ended(
                        sections=sections))
```
But as you have to calculate the variances, which are then not stored in the `DataStore`, you can't chain it for longer. I think it is more clear without chaining anyway.

---

Will the memory use be higher though, when comparing
```python
ds = ds.calibration_single_ended(inplace=False)
```
to
```python
ds.calibration_single_ended(inplace=True)
```

---

An example use case;
With `inplace=True`:
```python
ds1 = read_silixa_files(files)
sections_1 = {~}
sections_2 = {~}

ds2 = ds1.copy(deep=True)
ds1.calibration_single_ended(sections=sections_1)
ds2.calibration_single_ended(sections=sections_2)
```

With `inplace=False`:
```python
ds = read_silixa_files(files)
sections_1 = {~}
sections_2 = {~}

ds1 = ds.calibration_single_ended(sections=sections_1)
ds2 = ds.calibration_single_ended(sections=sections_2)
```

An added benefit with `inplace=False` is that you can try things out in the console or a code cell, without it overwriting the original `DataStore`.

**Comment by bdestombe** [comment]
Ah, with your final example I finally understood what you meant.

This is great! Yes, that would be a great addition! This means we would have to collect `data_vars` throughout the calibration routine, and depending on `inplace` either:

- add the `data_vars` to self and return nothing
- Or create and return a new DataStore that includes old and new `data_vars`?

**Comment by BSchilperoort** [comment]
I had a look how to see how xarray does it. For example the `set_index` method of `Dataset`;
```python

    def set_index(
        self,
        ~
        ):

        indexes = either_dict_or_kwargs(indexes, indexes_kwargs, "set_index")
        variables, coord_names = merge_indexes(
            indexes, self._variables, self._coord_names, append=append
        )
        return self._replace_vars_and_dims(variables, coord_names=coord_names)
```
Their implementation is to not modify `self`, and return an object with the data replaced. This could be toggled with an `inplace` kwarg.

As this is more a style-thing and does nothing but change the syntax, this could be a nice, but low priority, addition for future development.

**Comment by bdestombe** [comment]
Shall we move to inplace=False when we construct the accessor?

**Comment by BSchilperoort** [comment]
Fixed by #204
